ethereum/core/src/main/java/org/hyperledger/besu/ethereum/worldstate/WorldStateStorageCoordinator.java

```diff
@@ -62,14 +62,23 @@ public Optional<Bytes> getAccountStorageTrieNode(
         forest -> forest.getAccountStorageTrieNode(nodeHash));
   }
 
+  public Optional<Bytes> getCode(final Hash codeHash, final Hash accountHash) {
+    return applyForStrategy(
+        bonsai -> bonsai.getCode(codeHash, accountHash), forest -> forest.getCode(codeHash));
+  }
+
   @SuppressWarnings("unchecked")
   public <STRATEGY extends WorldStateKeyValueStorage> STRATEGY getStrategy(
       final Class<STRATEGY> strategyClass) {
     return (STRATEGY) worldStateKeyValueStorage;
   }
 
   public boolean isMatchingFlatMode(final FlatDbMode flatDbMode) {
-    if (getDataStorageFormat().equals(DataStorageFormat.BONSAI)) {
+    Optional<DataStorageFormat> storageFormat =
+        Optional.ofNullable(getDataStorageFormat())
+            .filter(format -> format.equals(DataStorageFormat.BONSAI));
```

Comment:
why we need that ?

Reply:
IIRC there are a lot of test cases that do not fully setup the coordinator, and datastorageformat was coming back nullable.  I will remove this Optional and see which tests fail and fix those.